core-common/src/main/java/org/zalando/nakadi/service/AvroSchema.java

```diff
@@ -7,41 +7,70 @@
 import org.springframework.beans.factory.annotation.Value;
 import org.springframework.core.io.Resource;
 import org.springframework.stereotype.Service;
+import org.zalando.nakadi.exceptions.runtime.NoSuchEventTypeException;
+import org.zalando.nakadi.exceptions.runtime.NoSuchSchemaException;
 import org.zalando.nakadi.util.AvroUtils;
 
+import java.io.InputStream;
 import java.io.IOException;
+import java.util.Comparator;
+import java.util.Map;
+import java.util.HashMap;
+import java.util.Set;
+import java.util.TreeMap;
 
 // temporarily storage for event type avro schemas untill schema repository supports them
 @Service
 public class AvroSchema {
 
     public static final byte METADATA_VERSION = 0;
+    public static final Comparator<String> SCHEMA_VERSION_COMPARATOR = Comparator.comparingInt(Integer::parseInt);
+
     private final Schema metadataSchema;
-    private final Schema nakadiAccessLogSchema;
+    private final Map<String, TreeMap<String, Schema>> eventTypeSchema;
     private final AvroMapper avroMapper;
     private final ObjectMapper objectMapper;
 
     @Autowired
     public AvroSchema(
             final AvroMapper avroMapper,
             final ObjectMapper objectMapper,
-            @Value("${nakadi.avro.schema.metadata:classpath:metadata.avsc}")
-            final Resource metadataRes,
-            @Value("${nakadi.avro.schema.nakadi-access-log:classpath:nakadi.access.log.avsc}")
-            final Resource nakadiAccessLogRes)
+            @Value("${nakadi.avro.schema.metadata:classpath:event-type-schema/metadata.avsc}")
+            final Resource metadataSchemaRes,
+            @Value("${nakadi.avro.schema.internal-event-types:classpath:event-type-schema/}")
+            final Resource eventTypeSchemaRes)
             throws IOException {
         this.avroMapper = avroMapper;
         this.objectMapper = objectMapper;
-        this.metadataSchema = AvroUtils.getParsedSchema(metadataRes.getInputStream());
-        this.nakadiAccessLogSchema = AvroUtils.getParsedSchema(nakadiAccessLogRes.getInputStream());
-    }
+        this.metadataSchema = AvroUtils.getParsedSchema(metadataSchemaRes.getInputStream());
+        this.eventTypeSchema = new HashMap<>();
 
-    public Schema getNakadiAccessLogSchema() {
-        return nakadiAccessLogSchema;
+        for (final String eventTypeName : Set.of("nakadi.access.log")) {
```

Comment:
could we try to load every file from `event-type-schema` folder and load then into tree map

Reply:
This will require a bigger change to make this work nicely: e.g. moving these files from resources to container level.  Otherwise, listing resources within a jar within a jar seems to require a lot of effort.